nix/haskell.nix

```diff
@@ -1,52 +1,277 @@
-{ sourcesFile ? ./sources.json
-, system ? builtins.currentSystem
-, sources ? import ./sources.nix { inherit system sourcesFile; }
-, plutus ? import sources.plutus { }
-, plutus-apps ? import sources.plutus-apps { }
-, deferPluginErrors ? true
-, doCoverage ? false }:
-let inherit (plutus-apps) pkgs;
-    utils = import ./utils.nix;
-in pkgs.haskell-nix.cabalProject rec {
-  src = pkgs.haskell-nix.haskellLib.cleanGit {
-    name = "gero-gov";
-    src = ./..;
-  };
+{ src, inputs, pkgs, doCoverage ? false, deferPluginErrors ? true, ... }:
 
-  # Plutus uses a patched GHC. And so shall we.
-  compiler-nix-name = "ghc810420210212";
+pkgs.haskell-nix.cabalProject {
+  inherit src;
 
-  # -- Materialization
-  # See https://input-output-hk.github.io/haskell.nix/tutorials/materialization/:
-  # Update using:
-  #   nix-build default.nix 2>&1 | grep -om1 '/nix/store/.*-updateMaterialized' | bash
-  # plan-sha256 = "0m56bhk9w3v1zqpig84f9krrp6sqg21w0vxbjiqcxz8n7c39aw54";
-  # materialized = ./materialization/gero-gov.materialized;
+  name = "plutus-simple-model";
 
-  modules = [{
-    packages = {
-      cardano-crypto-praos.components.library.pkgconfig =
-        pkgs.lib.mkForce [ [ pkgs.libsodium-vrf ] ];
+  compiler-nix-name = "ghc8107";
 
-      cardano-crypto-class.components.library.pkgconfig =
-        pkgs.lib.mkForce [ [ pkgs.libsodium-vrf ] ];
+  shell = {
+    inputsFrom = [ pkgs.libsodium-vrf ];
+
+    # Make sure to keep this list updated after upgrading git dependencies!
+    additional = ps:
+      with ps; [
+        filemanip
```

Comment:
You don't need filemanip and ieee

Reply:
If you have any reference to them in `cabal.project`, you should remove the references. The `cabal.project` doesn't really need to have anything at all besides `allow-newer: size-based:template-haskell`.